packages/medusa/src/api/routes/admin/product-categories/index.ts

```diff
@@ -36,47 +55,49 @@ export default (app) => {
 
   route.post(
     "/",
-    transformQuery(AdminPostProductCategoriesParams, {
-      defaultFields: defaultProductCategoryFields,
-      defaultRelations: defaultAdminProductCategoryRelations,
-      isList: false,
-    }),
+    transformQuery(
+      AdminPostProductCategoriesParams,
+      atomicTransformQueryConfig
+    ),
     transformBody(AdminPostProductCategoriesReq),
     middlewares.wrap(createProductCategory)
   )
 
   route.get(
     "/",
-    transformQuery(AdminGetProductCategoriesParams, {
-      defaultFields: defaultProductCategoryFields,
-      defaultRelations: defaultAdminProductCategoryRelations,
-      isList: true,
-    }),
+    transformQuery(AdminGetProductCategoriesParams, listTransformQueryConfig),
     middlewares.wrap(listProductCategories)
   )
 
   route.get(
     "/:id",
-    transformQuery(AdminGetProductCategoryParams, {
-      defaultFields: defaultProductCategoryFields,
-      isList: false,
-    }),
+    transformQuery(AdminGetProductCategoryParams, atomicTransformQueryConfig),
     middlewares.wrap(getProductCategory)
   )
 
   route.post(
     "/:id",
-    transformQuery(AdminPostProductCategoriesCategoryParams, {
-      defaultFields: defaultProductCategoryFields,
-      defaultRelations: defaultAdminProductCategoryRelations,
-      isList: false,
-    }),
+    transformQuery(
+      AdminPostProductCategoriesCategoryParams,
+      atomicTransformQueryConfig
+    ),
     transformBody(AdminPostProductCategoriesCategoryReq),
     middlewares.wrap(updateProductCategory)
   )
 
   route.delete("/:id", middlewares.wrap(deleteProductCategory))
 
+  route.post(
+    "/:id/products/batch",
+    transformQuery(
+      AdminPostProductCategoriesCategoryProductsBatchParams,
+      atomicTransformQueryConfig
+    ),
+    transformBody(AdminPostProductCategoriesCategoryProductsBatchReq),
+    validateProductsExist((req) => req.body.product_ids),
```

Comment:
though: not related to your pr, but this middleware is tight to the sales channels by his typings of the getProduct. should we just expect the return collection to be a partial product collection instead. also the middleware contains a lot of `sc` and `scId` where it should be `product` and `productId`

Reply:
Good find. I'll refactor it.